### PR TITLE
New Flashable store for vecengine

### PR DESCRIPTION
**Pull request by arrivets**
This PR implements a faster store for the `vecengine`.

### TL;DR
It is a wrapper around a native map that uses a backup db when its size exceeds a certain limit.
It is only intended for use with `vecengine`.
It uses stubs to implement the full kvdb.Store interface, but will panic if calling some functions like `Delete`, `NewIterator` or `NewBatch`.

### Overview
The core vector engine (pkg vecengine) in lachesis-base currently uses the `Flushable` wrapper (pkg flushable) around a provided database, to enable discarding db transactions when a certain operation fails. One of the problems this PR addresses is that `Flushable` does a lot more than what is strictly required by vecengine, and is slower than necessary. Hence, we implemented the `Flashable` wrapper, with less features than `Flushable`, which uses a native map instead of red-black trees, among other optimisations. The second problem that we address, is that in the current implementation of go-opera, the database that is wrapped by the vecengine’s Flushable is a LevelDB database which is slow because it reads and writes from disk. Our Flashable wrapper, uses in-memory storage with a backup mechanism that unloads data onto a backing database when its size exceeds a certain limit. The idea being that vecengine should use in memory storage only, but still have a backup security mechanism in case it gets too big.

### Main points about Flashable.

- Add only (no individual deletion, although there is a way to clear all the data at once)
- No Iterator
- No Batch
- Using native maps instead of red-black trees
- Data is unloaded to an underlying database when the size of the store reaches a preconfigured limit. 
- Not all data is unloaded at once, but only the “bottom” half. Older items are unlikely to be requested, so it’s ok to move them to persistent db.

### CODE REVIEW

**Comment by uprendis** [comment]
Squished commits (because [af1fe5c](https://github.com/Fantom-foundation/lachesis-base/pull/58/commits/af1fe5c6777fd9af5ad44948ffdec97ee58397bb) contained large binary files) & refactored: #73, #74
